src/posix/platform/hdlc_interface.hpp

```diff
@@ -158,16 +158,16 @@ class HdlcInterface
     uint32_t GetBusSpeed(void) const { return mBaudRate; }
 
     /**
-     * This method is called when RCP failure detected and resets internal states of the interface.
+     * This method resets the RCP.
      *
-     */
-    void OnRcpReset(void);
-
-    /**
-     * This method is called when RCP is reset to recreate the connection with it.
+     * @param[in] aResetType The reset type.
+     *
+     * @retval OT_ERROR_NONE            Successfully reset the RCP.
+     * @retval OT_ERROR_FAILED          Failed to reset the RCP.
+     * @retval OT_ERROR_NOT_IMPLEMENT   The specified reset type is not implemented.
      *
      */
-    otError ResetConnection(void);
+    otError Reset(uint8_t aResetType);
```

Comment:
As the radio_spinel already has the ability to do software reset, i don't think we need provide parameter for this method. We should keep interface orthogonal.